src/browser/util/encodeArgs.js

```diff
@@ -0,0 +1,30 @@
+import { u, wallet } from '@cityofzion/neon-js';
+import tree from 'switch-tree';
+
+const encodeArgs = (args) => {
+  // assert args is an array
+  if (!Array.isArray(args)) {
+    return args;
+  }
+
+  // For now this takes care of strings and numbers and nested Arrays
+  // Also this shouldn't be used when NEO addresses are passed for contracts using CheckWitness.
+  // Until we determine what a good practice for general encoding addresses is
+  // - using unhexlify + reverseHex + toScriptHash or just reverseHex as is used in this function -
+  // but the former is necessary for contracts using CheckWitness
+  // the latter is used from neon-js guides to checkBalance.
+  // If the former can be used for those as well, this is no problem.
+  const encodedArgs = args.map((arg) => {
+    const executor = tree`
+      lazy ${arg}
+        expression ${wallet.isAddress} ${() => u.reverseHex(wallet.getScriptHashFromAddress(arg))}
+        expression ${Array.isArray} ${() => encodeArgs(arg)}
+        type ${'string'} ${() => u.str2hexstring(arg)}
+        type ${1} ${() => u.int2hex(arg)}
```

Comment:
It's not clear to me here (even after reading the package documentation) why you call `type ${1}` here instead of something like `type ${'number'}`.  Would anything change here if you called `type ${2}` instead?  Maybe I'm interpreting it as the equivalent of JS's `typeof` operator when that's not the case...?

Reply:
The `type` evaluator does typeof x === typeof y. So in this case you're looking for a number, you have to evaluate it against a number. But that's a good suggestion though, I might change it in a later version.